src/functional/ffront/ast_passes/single_static_assign.py

```diff
@@ -88,6 +104,62 @@ def __init__(self, separator="__"):
     def _rename(self, node):
         return self.RhsRenamer.apply(self.name_counter, self.separator, node)
 
+    def visit_FunctionDef(self, node: ast.FunctionDef):
+
+        # For practical purposes, this is sufficient, but really not general at all.
+        # However, the algorithm was never intended to be general.
+
+        pre_assigns: list[ast.stmt] = [
+            _make_assign(arg.arg, arg.arg, arg) for arg in node.args.args
+        ]
+        # This forces versioning for parameters which is required for if-stmt support
+        node.body = pre_assigns + node.body
+        node.body = [self.visit(stmt) for stmt in node.body]
```

Comment:
```suggestion
```
I would propose to avoid this and instead use the original name of the variable in the `visit_Name` and `visit_If` in case its name_counter is zero.

Reply:
Fixed this in the latest version